### ISSUE TITLE
[feat] Handle CLI arguments and subcommands

### ISSUE THREAD

**Issue by rochacbruno** [rochacbruno/marmite#1]
Right now only `marmite .` is working.

What is wanted:

```bash
marmite [input_folder] [output_folder] (--options)
```

```console
$ marmite myblog site  # build the site
$ martite myblog site --serve # Build and Serve with embedded http in rebuild mode
```

A `--config foo.yaml` should also be supported to customize the `marmite.yaml` file location.l, if not informed this file will be loaded from input_folder/marmite.yaml if exists, else default config will be loaded.

**Comment by scovl** [comment]
Is there a Golang-style Cobra framework in Rust?  Or would you prefer to build everything here and avoid external libs as much as possible? @rochacbruno

**Comment by rochacbruno** [comment]
We can use clap it is well stable and widely used.

I was thinking about following more the **unix** style CLI.

Build a site

```
marmite input_dir output_dir
```

Just like the common `cp foo bar` or `mv foo bar`

This will require some logic change.

- if a marmite.yml is found, it is used to override the default config, if not, continue with default config.
- If `input_folder` has a content subdir, the markdown from that folder is used
- If there is no `content` subdir, then all .md from the root of input folder is processed.
- If there is a templates/ folder those are rendered, if not, then default templates embedded in the binary are rendered.
- is static and media folders are found, those are copied to output_dir.


This way, marmite will always require 2 positional arguments, if executed from inside the input_folder then `.` can be used.

```
mkdir myblog
echo "#my page" > myblog/page.md
echo "#my post" > myblog/2024-01-31-my-post.md
```
then simply
```
marmite myblog site
```

On the site folder there will be a index.html listing "my post", a page.html and a my-post.html

all using the embedded templates.

**Comment by rochacbruno** [comment]
https://docs.rs/clap/latest/clap/

**Comment by rochacbruno** [comment]
Updated the README with the desired CLI https://github.com/rochacbruno/marmite?tab=readme-ov-file#usage

**Comment by scovl** [comment]
> Right now only `marmite .` is working.
> 
> What is wanted:
> 
> ```shell
> marmite [input_folder] [output_folder] (--options)
> ```
> 
> ```
> $ marmite myblog site  # build the site
> $ martite myblog site --serve # Build and Serve with embedded http in rebuild mode
> ```
> 
> A `--config foo.yaml` should also be supported to customize the `marmite.yaml` file location.l, if not informed this file will be loaded from input_folder/marmite.yaml if exists, else default config will be loaded.

I'm thinking about it...It still feels like there are too many subcommands. For example: `marmite myblog` seems enough to generate the blog and the init, so you wouldn't need a separate init command... It's intuitive. New, `marmite myblog site` (what would `site` be?) to render from markdown to HTML?

If that's the case, the second command `marmite myblog site --serve` doesn't seem to make sense either. it could simply be `marmite --serve`, which would operate within the myblog directory.

Resume:

$ marmite myblog  # build the site
$ marmite myblog  --serve # Build and Serve with embedded http in rebuild mode

With `--serve`, It would start the web service and render the markdown content into HTML. Maybe...it would be interesting to add a `--render` parameter here (just to render without needing to start the web service). For cases where you already know what the output will be and just want to upload it to hosting. I think the term `render` or `--render` is more intuitive than `site`. The word `site` is way too generic.

**Comment by rochacbruno** [comment]
I was thinking about having `marmite` to be as simple as `cp` or `mv` linux commands.

So the idea is that it will work, without any configuration, the only requirement is a folder full of markdown files on its root. (we can think about traversing nested folders later)

Consider the following workflow:


There is a folder full of `md` files, does not matter if the files includes the frontmatter metadata or not, the site will include it.

```plain
mymarkdowns
├── about.md
├── first-blog-post.md
└── 2024-01-01-second-blog-post.md
```

All the user needs to do is 

```console
$ marmite mymarkdowns mysite
Generated index.html
Generated pages.html
Generated second-blog-post.html
Generated blog-post.html
Generated about.html
Site generated at: mysite/
```

All the user has to do now is to take the `mysite` folder and deploy it.


## Optional config

Optionally user can add a `mymarkdowns/marmite.yaml` to customize build options.

## Optional folder structure

Optionally the user can have all markdowns on `mymarkdowns/content` folder, for better organization.

## Template Customization

Optionally user can add a `mymarkdowns/templates`, `mymarkdown/static` to customize look & feel of the website. If not provided, marmite will use embedded templates (#20)

## Media management

If user wants to add local media to the website, the files must go to `mymarkdowns/media` folder and that folder will be copied to the generated site when building.

## Serving during writing or Template customization

User can use `--serve` to start marmite built-in http server that will serve the site and rebuild it on every file change under `mymarkdowns` folder.

## So

It is important to keep the semantics `marmite input_folder output_folder` 

The only options will be `--serve` and `--config`

The `init` command is not needed, user should create folder structure using common terminal tools like `mkdir` etc...

**Comment by scovl** [comment]
Interesting! Now I understand it better!
It's as simple and minimalistic as possible!

**Comment by rochacbruno** [comment]
resolved on #30
